smarts/env/gymnasium/wrappers/metric/completion.py

```diff
@@ -133,16 +133,21 @@ def _dist_remainder():
     step: int = 0
 
     def func(
-        road_map: RoadMap, obs: Dict[str, Any], initial_compl: Completion
+        road_map: RoadMap, obs: Observation, initial_compl: Completion
     ) -> Completion:
         nonlocal mean, step
 
-        if obs["events"]["reached_goal"]:
+        if obs.events.reached_goal:
             dist = 0
         else:
-            cur_pos = Point(*obs["ego_vehicle_state"]["position"])
-            goal_pos = Point(*obs["ego_vehicle_state"]["mission"]["goal_position"])
-            dist = get_dist(road_map=road_map, point_a=cur_pos, point_b=goal_pos)
+            cur_pos = Point(*obs.ego_vehicle_state.position)
+            goal_pos = Point(
+                *getattr(obs.ego_vehicle_state.mission.goal, "position", None)
+            )
+            if goal_pos is not None:
+                dist = get_dist(road_map=road_map, point_a=cur_pos, point_b=goal_pos)
+            else:
+                dist = 0
```

Comment:
`goal_pos` cannot be `None` as we check each scenario's suitability, i.e., availability of `PositionalGoal`.

https://github.com/huawei-noah/SMARTS/blob/84637f4c185f2c105d8ea5a7c56a0d8ec13ff5d7/smarts/env/gymnasium/wrappers/metrics.py#L358-L375

Reply:
I have changed it to assertions. The issue here is that the type requirement is not indicated through the interface.

If the method required a position or a positional goal as part of the arguments that would be possible but this method has no relation to the previous goal constraint.